set8/58/numbers/operatorassign2.cc

```diff
@@ -0,0 +1,12 @@
+#include "numbers.ih"
+
+    // by 
+
+Numbers &Numbers::operator=(Numbers &&tmp)
+{
+    swap(tmp);
+    delete[] tmp.d_data;                    // Manually clearing to prevent
+    tmp.d_size = 0;                         // tmp holding "this" object's
+    tmp.d_data = nullptr;                   // contents after swap
+    return *this;                           
```

Comment:
I assume you do this because of the comment in the destructor when using `makeclass`:

```cpp
    // move operations must allow the destructor to decide that no
    // destruction should be performed, e.g. by setting a pointer data
    // member to 0, returning immediately if this data member equals 0
```

Manually clearing here is superfluous if you don't shortcut the destructor. The `tmp` object will be destroyed after this scope so the destructor will be called, which still calls `delete[] tmp.d_data`. If you don't shortcut in the destructor (by checking that, e.g., `d_data = nullptr`) then effectively you destroy twice, with the latter being a no-op because it's deleting a nullptr.

Reply:
(I personally would let the destructor handle everything without shortcutting at all.)